src/plugins/workspace/public/components/workspace_list/index.tsx

```diff
@@ -361,124 +442,151 @@ export const WorkspaceList = ({ registeredUseCases$ }: WorkspaceListProps) => {
         return renderDataWithMoreBadge(dataSources, 2, item.id, DetailTab.DataSources);
       },
     },
+  ];
+  const allActions = [
+    {
+      name: <EuiText key="copyId">Copy ID</EuiText>,
+      icon: 'copy',
+      type: 'icon',
+      description: 'Copy workspace id',
+      'data-test-subj': 'workspace-list-copy-id-icon',
+      onClick: ({ id }: WorkspaceAttribute) => handleCopyId(id),
+    },
+    {
+      name: <EuiText key="edit">Edit</EuiText>,
+      icon: 'pencil',
+      type: 'icon',
+      description: 'Edit workspace',
+      'data-test-subj': 'workspace-list-edit-icon',
+      onClick: ({ id }: WorkspaceAttribute) => handleSwitchWorkspace(id),
+    },
+    {
+      name: <EuiText key="setDefault">Set as my default</EuiText>,
+      icon: 'flag',
+      type: 'icon',
+      description: 'Set as default workspace',
+      'data-test-subj': 'workspace-list-set-default-icon',
+      onClick: (item: WorkspaceAttribute) => handleSetDefaultWorkspace(item),
+    },
+    {
+      name: <EuiText key="leave">Leave</EuiText>,
+      icon: 'exit',
+      type: 'icon',
+      description: 'Leave workspace',
+      'data-test-subj': 'workspace-list-leave-icon',
+      available: () => false,
```

Comment:
Do we need this config if it would always be unavailable?

Reply:
we plan to enable it in next PR to let user leave from selected workspace only if user permission granted by individual user id not group.